tests/Io/ConnectionTest.php

```diff
@@ -278,14 +276,12 @@ public function testConnectWithValidPass()
         $conn->doConnect(function ($err, $conn) use ($loop) {
             $this->assertEquals(null, $err);
             $this->assertInstanceOf('React\MySQL\Io\Connection', $conn);
-            $this->assertEquals(Connection::STATE_AUTHENTICATED, $conn->getState());
         });
 
-        $conn->ping()->then(function () use ($loop, $conn) {
-            $conn->quit()->done(function () use ($conn) {
-                $this->assertEquals($conn::STATE_CLOSED, $conn->getState());
-            });
+        $conn->ping()->then(function () use ($conn) {
+            $conn->quit()->then($this->expectCallableOnce());
         });
+
```

Comment:
The creation of the mock should be done outside the callback to ensure it is tracked by PHPUnit.

```php
$quitMock = $this->expectCallableOnce();

$conn->ping()->then(function () use ($conn, $quitMock) {
    $conn->quit()->then($quitMock);
});
```

Reply:
Good catch, updated :+1: